lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +434,1140 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+    FPM.addPass(RemoveHelperCallsPass());
+    FPM.addPass(PromoteGlobalToLocalPass());
+    FPM.addPass(SimplifyCFGPass());
+    FPM.addPass(SROA());
+    FPM.addPass(EarlyCSEPass(true));
+    FPM.addPass(JumpThreadingPass());
+    FPM.addPass(UnreachableBlockElimPass());
+    FPM.addPass(InstCombinePass(true));
+    FPM.addPass(EarlyCSEPass(true));
+    FPM.addPass(SimplifyCFGPass());
+    FPM.addPass(MergedLoadStoreMotionPass());
+    FPM.addPass(GVN());
+
+    // Second stage: add alias analysis info and canonicalize i2p + add into
+    // geps. Since the IR may change remarkably, another round of passes is
+    // necessary to take more optimization opportunities.
+    FPM.addPass(SegregateDirectStackAccessesPass());
+    FPM.addPass(EarlyCSEPass(true));
+    FPM.addPass(InstCombinePass(true));
+    FPM.addPass(GVN());
+
+    // Third stage: if enabled, serialize the results and dump the functions on
+    // disk with the alias information included as comments.
+    if (IndirectBranchInfoSummaryPath.getNumOccurrences() == 1)
+      FPM.addPass(IndirectBranchInfoPrinterPass(OutputIBI));
+
+    if (AAWriterPath.getNumOccurrences() == 1)
+      FPM.addPass(AAWriterPass(OutputAAWriter));
+
+    FunctionAnalysisManager FAM;
+    FAM.registerPass([] {
+      AAManager AA;
+      AA.registerFunctionAnalysis<BasicAA>();
+      AA.registerFunctionAnalysis<ScopedNoAliasAA>();
+
+      return AA;
+    });
+
+    ModuleAnalysisManager MAM;
+    MAM.registerPass([&] { return GeneratedCodeBasicInfoAnalysis(); });
+    FAM.registerPass([&] { return GeneratedCodeBasicInfoAnalysis(); });
+    FAM.registerPass([&] { return ModuleAnalysisManagerFunctionProxy(MAM); });
+
+    PassBuilder PB;
+    PB.registerFunctionAnalyses(FAM);
+    PB.registerModuleAnalyses(MAM);
+
+    FPM.run(*OutlinedFunction, FAM);
+  }
+
+  // Restore value back.
+  getUOpt(Options, MemSSALimit)->setInitialValue(MemSSALimitDefault);
+  getUOpt(Options, MemDepBlockLimit)->setInitialValue(MemDepBlockLimitDefault);
+
+  FunctionSummary FunctionInfo = milkInfo(ABIRegisters,
+                                          ABIResults,
+                                          WrittenRegisters,
+                                          CFG,
+                                          OutlinedFunction);
+
+  // Is the disposable function a FakeFunction? If so, return a
+  // copy of the unoptimized disposable function.
+  if (FunctionInfo.Type == FunctionTypeValue::Fake) {
+    FunctionInfo.FakeFunction = createDisposableFunction(Entry);
+
+    if (AnyPCCloned) {
+      SmallVector<Instruction *, 4> BrokenReturns;
+      for (auto *Pred : predecessors(AnyPCCloned))
+        BrokenReturns.emplace_back(Pred->getTerminator());
+
+      for (auto *Br : BrokenReturns) {
+        auto *Ret = ReturnInst::Create(Context);
+        ReplaceInstWithInst(Br, Ret);
+      }
+    }
+
+    if (UnexpectedPCCloned) {
+      auto *FTy = llvm::FunctionType::get(Type::getVoidTy(Context), false);
```

Comment:
This function can be create only once for the whole run.
Also, remember to erase all the temporary functions you create (unless they come from an `OpaqueFunctionPool` *and* you instructed it to destroy them).